pkg/resource/meta.go

```diff
@@ -39,3 +39,15 @@ func buildPodObjectMeta(instance *v1beta1.TemporalCluster, service string) metav
 		),
 	}
 }
+
+// buildPodObjectMeta return ObjectMeta for app workers.
+func buildAppWorkerPodObjectMeta(instance *v1beta1.TemporalAppWorker, service string) metav1.ObjectMeta {
```

Comment:
IMO this function is not needed, you can use the `buildPodObjectMeta` for the app worker too :)

Reply:
@alexandrevilain again its a casting issue, we pass in TemporalCluster so I cannot pass in TemporalWorker process unless I pass an interface but then imho that doesnt make it nicer just more complex.